core/org.wso2.carbon.utils/src/main/java/org/wso2/carbon/utils/DiagnosticLog.java

```diff
@@ -51,6 +61,24 @@ public DiagnosticLog(String logId, Instant recordedAt, String requestId, String
         this.componentId = componentId;
         this.input = input;
         this.configurations = configurations;
+        this.logLevel = LogLevel.BASIC;
+    }
+
+    public DiagnosticLog(String logId, Instant recordedAt, String requestId, String flowId,
+            String resultStatus, String resultMessage, String actionId, String componentId,
+            Map<String, Object> input, Map<String, Object> configurations, LogLevel logLevel) {
+
+        this.logId = logId;
+        this.recordedAt = recordedAt;
+        this.requestId = requestId;
+        this.flowId = flowId;
+        this.resultStatus = resultStatus;
+        this.resultMessage = resultMessage;
+        this.actionId = actionId;
+        this.componentId = componentId;
+        this.input = input;
+        this.configurations = configurations;
+        this.logLevel = logLevel;
     }
```

Comment:
shall we change the new public constructor to use a static nested builder class?

eg: public DiagnosticLog(PublicDiagnosticLogEventBuilder builder)

Reply:
Fixed with a3cd4390c2d8ceeaeb711f9a6580e000ceb7c4b7 and also make the constructor private as well